cpp/util/etcd.cc

```diff
@@ -573,9 +573,8 @@ void EtcdClient::WatchRequestDone(WatchState* state, GetResponse* get_resp,
     return;
   }
 
-  // This is probably due to a timeout, just retry.
-  if (!child_task->status().ok()) {
-    LOG_EVERY_N(INFO, 10) << "Watch request failed: " << child_task->status();
+  if (child_task->status().error_code() == util::error::DEADLINE_EXCEEDED) {
+    VLOG(1) << "Watch request timedout: " << child_task->status();
```

Comment:
I restricted this to timeouts-only based on the comment that was there, but we could extend that to other errors if that was the right thing to do.  I don't really have a strong opinion either way other than fear of hiding a permanently borked etcd from the higher layers.

This change was mainly intended to clear up the misleading `UNAVAILABLE` coming from the read timeout on the hanging get.